tests/shared/drive-constraints.spec.js

```diff
@@ -556,4 +556,266 @@ describe('Shared: DriveConstraints', function() {
 
   });
 
+  describe('.getDriveImageCompatibilityStatus', function() {
+
+    beforeEach(function() {
+      if (process.platform === 'win32') {
+        this.mountpoint = 'E:';
+        this.separator = '\\';
+      } else {
+        this.mountpoint = '/mnt/foo';
+        this.separator = '/';
+      }
+
+      this.drive = {
+        device: '/dev/disk2',
+        name: 'My Drive',
+        protected: false,
+        system: false,
+        mountpoints: [
+          {
+            path: this.mountpoint
+          }
+        ],
+        size: 4000000000
+      };
+    });
+
+    describe('given there are no errors or warnings', () => {
+
+      it('should return OK', function() {
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
```

Comment:
Hmmm, I guess using `.to.deep.equal` means that all the unit-tests will instantly break if we want to add any extra attributes the the status-objects. Would a more flexible approach be to check:
```
m.chai.expect(result.type).to.equal(constraints.COMPATIBILITY_STATUS_TYPES.OK);
m.chai.expect(result.message).to.equal(constraints.COMPATIBILITY_STATUS_MESSAGES.OK);
```
?  \*shrug\*